### DISCUSSION TITLE
Upward JS or upward-php ?

### DISCUSSION THREAD

**Discussion by jerschipper** [magento/pwa-studio#3172] [General]
So, for the project i'm working on we're using the upward-php-connector. It works fine, but can somebody please help me explain what the difference is from both implementations?  Is it best practice to use Upward-js with a tooling like pm2? Or is the upward-php module just fine? 

Would like to hear what you guys are using and what your experience is with your current implementation :)

**Comment by jcalcaben** [comment]
The main difference between the two implementations is the language they are written in. Both are valid UPWARD servers that behave the same if you give them the same configuration file. 

Using the UPWARD-PHP implementation is more common if you are hosting your storefront code on the same server as your Magento backend instance (which is the case when using the Adobe Commerce Cloud service).This setup requires you to install the `upward-php-connector` module which will also include the `upward-php` server because it is a dependency.

The purpose of the `upward-php-connector` is to replace the default Magento frontend with your PWA Studio storefront. It does this by routing requests for the default Magento theme to the UPWARD-PHP server.

For more information about this setup, see the [cloud deployment tutorial](https://magento.github.io/pwa-studio/tutorials/cloud-deploy/).

UPWARD-JS is also a valid solution for production if your service provider supports running node processes.
